src/test/tree/StrategyManangerUnit.tree

```diff
@@ -0,0 +1,89 @@
+StrategyManagerUnit.t.sol
+├── initialize
+│   └── when initialize called again
+│       └── it should revert
+├── depositIntoStrategy()
+│   ├── when deposit successfully
+│   │   └── shares should increase and stakerStrategyListLength should increase (deposit new strategy)
+│   ├── when deposit successfully twice
+│   │   └── shares should increase and stakerStrategyListLength should increase (deposit new strategy)
+│   ├── when deposits paused
+│   │   └── it should revert
+│   ├── when re-entering
+│   │   └── it should revert
+│   ├── when safeTransferFrom() function reverts
+│   │   └── it should revert
+│   ├── when IERC20 token does not exist
+│   │   └── it should revert
+│   ├── when strategy.deposit() function reverts
+│   │   └── it should revert
+│   ├── when strategy does not exist
+│   │   └── it should revert
+│   ├── when strategy is not whitelisted
+│   │   └── it should revert
```

Comment:
technically the whitelisting feature can be deactivated entirely, so this should probably be 'when the whitelist is active and the strategy is not whitelisted' ?

Reply:
Active as in when deposits are not paused right? From the onlyWhenNotPaused(PAUSED_DEPOSITS) modifier. 
Then it could be 'when deposits are not paused and strategy is not whitelisted'

Reply:
ah I remembered wrong, I thought there was literally a 'whitelist enabled/disabled' kind of switch -- guess we decided an upgrade would be the appropriate place to remove the whitelisting functionality, if it ever happens